### PR TITLE
feature/SOF-3729

### PR DESCRIPTION
**Pull request by mohammadidinani**

### CODE REVIEW

File: src/data/handlers/exabyte.py [hunk 1]
```diff
@@ -148,12 +150,38 @@ def upload_files(self, job_id):
...
 
+    def upload_stdout(self, job_id):
+        """
+        Uploads stdout file.
+
+        Args:
+            job_id (str): job ID.
+        """
+        headers = self.job_endpoints.headers
+        endpoint_path = '/'.join(("jobs", job_id, "output-files"))
```
Comment (timurbazhirov) [inline src/data/handlers/exabyte.py]: Let's use `path.join`?

File: src/workflow/units/execution/espresso.py [hunk 2]
```diff
@@ -13,6 +18,22 @@ class EspressoExecutionUnit(BaseExecutionUnit):
...
+
+    @property
+    def version(self):
+        """
+        Returns the application version used in the unit.
+
+        Returns:
+             str
+        """
+        return "5.4.0"
```
Comment (timurbazhirov) [inline src/workflow/units/execution/espresso.py]: I don't understand the logic - why do we assume it's always that version? We can either set it to None/NotImplemented/Unknown or extract the version from output ...

File: src/workflow/units/execution/shell.py [hunk 3]
```diff
@@ -24,9 +24,19 @@ def application(self):
...
 
+    @property
+    def version(self):
+        """
+        Returns the application version used in the unit.
+
+        Returns:
+             str
+        """
+        return "4.2.46"
```
Comment (timurbazhirov) [inline src/workflow/units/execution/shell.py]: Less important for bash/shell, but the same point as above

File: src/workflow/units/execution/vasp.py [hunk 4]
```diff
@@ -30,10 +30,20 @@ def application(self):
...
 
+    @property
+    def version(self):
+        """
+        Returns the application version used in the unit.
+
+        Returns:
+             str
+        """
+        return "5.3.5"
```
Comment (timurbazhirov) [inline src/workflow/units/execution/vasp.py]: ++

File: src/data/handlers/exabyte.py (hunk 1 above)
Comment (mohammadidinani) [reply src/data/handlers/exabyte.py]: Done.

File: src/workflow/units/execution/espresso.py (hunk 2 above)
Comment (mohammadidinani) [reply src/workflow/units/execution/espresso.py]: Fixed.

File: src/workflow/units/execution/shell.py (hunk 3 above)
Comment (mohammadidinani) [reply src/workflow/units/execution/shell.py]: Fixed.

File: src/workflow/units/execution/vasp.py (hunk 4 above)
Comment (mohammadidinani) [reply src/workflow/units/execution/vasp.py]: Fixed.

**Pull request merged**